src/frontend/components/commentCard/CommentCard.js

```diff
@@ -0,0 +1,177 @@
+import { AvatarWithName } from "../../components";
+import { getUserDetails, updateComment } from "../../services";
+import { useSelector } from "react-redux";
+import { Input, Button } from "@chakra-ui/react";
+import { useState, useEffect, useRef } from "react";
+import { useOutsideClickHandler } from "../../custom-hooks/OutsideClickHandler";
+import "./CommentCard.css";
+
+const CommentCard = ({
+  comment,
+  reply,
+  editComment,
+  deleteComment,
+  increaseLikeCount,
+  decreaseLikeCount,
+}) => {
+  let data = comment ? comment : reply ? reply : "";
+  const votes = data?.votes > 0 ? data.votes : "";
+  const [likeState, setLikeState] = useState(false);
+  const { userDetails } = useSelector((state) => state.user);
+  const [openMenu, setOpenMenu] = useState(false);
+  const [displayReply, setReply] = useState(false);
+  const [displayEdit, setEdit] = useState(false);
+  const [commentText, setCommentText] = useState(data.content);
+  const [replyText, setReplyText] = useState("");
+  const ref = useRef();
+  const { resetMenu } = useOutsideClickHandler(ref);
+
+  useEffect(() => {
+    if (resetMenu) {
+      setOpenMenu(false);
+    }
+  }, [resetMenu]);
+
+  return (
+    <div className="comment-card flex-dir-col">
+      <div className="comment-nav flex-dir-row">
+        <AvatarWithName user={data.user} />
+        {!reply && data?.user?.username === userDetails?.username && (
+          <div className="ellipsis-icon" onClick={() => setOpenMenu(!openMenu)}>
+            <button className="comment-control">
+              <i class="fa fa-ellipsis-v" aria-hidden="true"></i>
+            </button>
+            {openMenu && (
+              <div className="comment-menu flex-dir-col" ref={ref}>
```

Comment:
In the Edit / Delete Modal, You can have the onclick for entire div rather than onclick on text

Reply:
sure